File: MitchEngine/Transform.cpp
#include "Transform.h"

using namespace ma;

Transform::Transform() : Position(glm::vec3(0,0,0)), Rotation(0), Scale(glm::vec3(1,1,1)){
}

Transform::~Transform() {
}

void Transform::Update(float dt) {

}

void Transform::Init() {

}

void Transform::SendMessage(class Message* message) {

}

File: Modules/ScriptCore/Source/Components/Transform.cs
namespace ScriptCore;

public class Transform : Component
{
    public Vector3 Position
    {
        get { unsafe { Vector3 v; Engine._api.Transform_GetTranslation(Entity, &v); return v; } }
        set { unsafe { Engine._api.Transform_SetTranslation(Entity, &value); } }
    }

    public Vector3 Scale
    {
        get { unsafe { Vector3 v; Engine._api.Transform_GetScale(Entity, &v); return v; } }
        set { unsafe { Engine._api.Transform_SetScale(Entity, &value); } }
    }

    // Euler
    public Vector3 Rotation
    {
        get { unsafe { Vector3 v; Engine._api.Transform_GetRotation(Entity, &v); return v; } }
        set { unsafe { Engine._api.Transform_SetRotation(Entity, &value); } }
    }
}
